homeassistant/components/amcrest/camera.py

```diff
@@ -365,10 +370,15 @@ def update(self):
                     self._brand = "unknown"
             if self._model is None:
                 resp = self._api.device_type.strip()
+                _LOGGER.debug("Device_type=%s", resp)
                 if resp.startswith("type="):
                     self._model = resp.split("=")[-1]
                 else:
                     self._model = "unknown"
+            if self._serial is None:
+                resp = self._api.serial_number.strip()
+                _LOGGER.debug("serial=%s", resp)
+                self._serial = resp
```

Comment:
I think if the serial number isn't available, for whatever reason, the return value here will be an empty string.

```suggestion
                self._serial = resp if resp else "unknown"
```

Reply:
With this adjustment, it will return `unknown` as a serial, why not leave serial `None`? That way, you don't need additional logic in the unique_id property method to filter out `unknown` again.

Reply:
That makes sense, `self._serial = resp if resp else None` would be good here.

Reply:
Not even needed, as the default is `None` already, additionally, serial isn't used for anything else.

So all can be replaced with a single 

```py
if resp:
  self._attr_unique_id = resp
```

Reply:
My only concern with this approach is if it can't get the serial number, how many times will the api query the device? If you set it to unknown after the first failure, it won't continue to retry. 

Initially I had hopes of having the camera motion sensor entity also have a unique id based on getting the serial from this device, but I gave up on that. If serial is an attribute then you could browse it, while the uniq id is intentionally not browsable.

Reply:
>  If you set it to unknown after the first failure, it won't continue to retry.

It needs to be set on creating of the entity, it cannot be provided at a later moment.

Reply:
Either way is fine with me - my tested setup is assigned it to the serial property and using that property if available for the unique id. Setting it directly is fine as well - you just can see it in the state browser.